gdal/frmts/raw/byndataset.cpp

```diff
@@ -0,0 +1,766 @@
+       eDT = GDT_Int16;
+    else
+        if ( poDS->hHeader.nSizeOf == 4 )
+            eDT = GDT_Int32;
+        else
+        {
+            delete poDS;
+            return nullptr;
+        };
+
+/* -------------------------------------------------------------------- */
+/*      Create band information object.                                 */
+/* -------------------------------------------------------------------- */
+
+    const int nDTSize = GDALGetDataTypeSizeBytes( eDT );
+
+    int bIsLSB = poDS->hHeader.nByteOrder == 1 ? 1 : 0;
+
+    BYNRasterBand *poBand = new BYNRasterBand(
+        poDS, 1, poDS->fpImage, BYN_HDR_SZ,
+        nDTSize, poDS->nRasterXSize * nDTSize, eDT, 
+        CPL_IS_LSB == bIsLSB );
+
+    poDS->SetBand( 1, poBand );
+
+/* -------------------------------------------------------------------- */
+/*      Initialize any PAM information.                                 */
+/* -------------------------------------------------------------------- */
+
+    poDS->SetDescription( poOpenInfo->pszFilename );
+    poDS->TryLoadXML();
+
+/* -------------------------------------------------------------------- */
+/*      Check for overviews.                                            */
+/* -------------------------------------------------------------------- */
+
+    poDS->oOvManager.Initialize( poDS, poOpenInfo->pszFilename );
+
+    return poDS;
+}
+
+/************************************************************************/
+/*                          GetGeoTransform()                           */
+/************************************************************************/
+
+CPLErr BYNDataset::GetGeoTransform( double * padfTransform )
+
+{
+    memcpy( padfTransform, adfGeoTransform, sizeof(double) * 6 );
+    return CE_None;
+}
+
+/************************************************************************/
+/*                          SetGeoTransform()                           */
+/************************************************************************/
+
+CPLErr BYNDataset::SetGeoTransform( double * padfTransform )
+
+{
+    if( padfTransform[2] != 0.0 || padfTransform[4] != 0.0 )
+    {
+        CPLError( CE_Failure, CPLE_AppDefined,
+                  "Attempt to write skewed or rotated geotransform to byn." );
+        return CE_Failure;
+    }
+    memcpy( adfGeoTransform, padfTransform, sizeof(double) * 6 );
+    return CE_None;
+}
+
+/************************************************************************/
+/*                          GetProjectionRef()                          */
+/************************************************************************/
+
+const char *BYNDataset::GetProjectionRef()
+
+{
+    OGRSpatialReference oSRS;
+
+    char* pszRefSystem = nullptr;
+
+    /* Try to use a prefefined EPSG compound CS */
+
+    if( hHeader.nDatum == 1 && hHeader.nVDatum == 2 ) 
+    {
+        oSRS.importFromEPSG( BYN_DATAM_1_VDATUM_2 );
+        oSRS.exportToWkt( &pszRefSystem );
```

Comment:
memory leak. The return value of GetProjectionRef() should be owned by the dataset